operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -416,4 +470,283 @@ private static Uri uri(FHIROperationContext operationContext, String uriPath) {
+            addSearchParameterIfNotExcluded(compartmentMemberType, "source", Practitioner.class.getSimpleName(), searchParameters);
+        } else if (compartmentMemberType.equals(Basic.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentMemberType, "author", Organization.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "author", Practitioner.class.getSimpleName(), searchParameters);
+        } else if (compartmentMemberType.equals(CarePlan.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentMemberType, "performer", Organization.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "performer", Practitioner.class.getSimpleName(), searchParameters); 
+        } else if (compartmentMemberType.equals(CareTeam.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentMemberType, "participant", Organization.class.getSimpleName(),searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "participant", Practitioner.class.getSimpleName(), searchParameters);
+        } else if (compartmentMemberType.equals("ChargeHistory")) {
+            addSearchParameterIfNotExcluded(compartmentMemberType, "service", Organization.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "enterer", Practitioner.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "performer-actor", Practitioner.class.getSimpleName(), searchParameters);
+        } else if (compartmentMemberType.equals(Claim.class.getSimpleName())) {
+            addSearchParameterIfNotExcluded(compartmentMemberType, "care-team", Organization.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "care-team", Practitioner.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "enterer", Practitioner.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "facility", Location.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "insurer", Organization.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "payee", Organization.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "payee", Practitioner.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "provider", Organization.class.getSimpleName(), searchParameters);
+            addSearchParameterIfNotExcluded(compartmentMemberType, "provider", Practitioner.class.getSimpleName(), searchParameters);
```

Comment:
FYI: and something you may want to consider.... a small optimization here would be to organize Practitioner, Organization then Location and pass in the allowedIncludes instead of multiple retrievals. 

you could change the method signature slightly to: `string, string, string...  ` and process the array of codes (nameOfSearchParameter)

Reply:
Done

Reply:
Done